src/gapstate.h

```diff
@@ -192,21 +194,45 @@ extern GAPState MainGAPState;
 
 #endif
 
-void InitMainGAPState(void);
+/* this is potentially confusing, because MODULE_STATE is supposed to be
+  *local* to the module, so MODULE_STATE will refer to a different bit of
+   state depending which file it is used in */
+#define MODULE_STATE(x) (((ModuleState *)(&STATE(StateSlots)[StateHandle]))->x)
+
+
+// Offset into StateSlots
+typedef Int ModuleStateHandle;
+typedef void (*ModuleConstructor)(ModuleStateHandle handle);
+typedef void (*ModuleDestructor)();
```

Comment:
In my mind the constructor gets the information about where to store state, and then has to store it if it wants to keep it, the destructor never needs this information (because the constructor stored it somewhere).

I am not opposed to adding the `handle` as a parameter to the destructor as well, I just think it'd not be used.

Reply:
Looking at the code just now, one could of course even leave the parameter off the constructor, since in the current usage-examples we store the offset when we call `RegisterModuleState`.